recommendation/views/images.py

```diff
@@ -0,0 +1,35 @@
+from urllib.parse import urlencode
+
+import requests
+from flask import abort, Blueprint, request, Response, stream_with_context
+
+from recommendation import conf
+
+EMBEDLY_RESIZE = 'https://i.embed.ly/1/display/resize'
+
+
+images = Blueprint('images', __name__)
+
+
+def make_embedly_url(**kwargs):
+    if 'url' not in kwargs:
+        raise RuntimeError()
+    qs = {}
+    for param in ['url', 'width', 'height']:
+        if param in kwargs:
+            qs[param] = kwargs[param][0]
+    qs['key'] = conf.EMBEDLY_API_KEY
+    return '{}?{}'.format(EMBEDLY_RESIZE, urlencode(qs))
+
+
+@images.route('/images')
+def proxy():
+    try:
+        url = make_embedly_url(**request.args)
+    except RuntimeError:
+        abort(400)
+    req = requests.get(url, stream=True)
+    stream = stream_with_context(req.iter_content())
```

Comment:
This should check for the response status code and only try to get the content if we got a 200 OK response. I couldn't find a clear explanation of all the possible error codes for embedly, but it seems they can at least return various 4xx responses. For example if the API key is invalid or if a url is requested that either doesn't exist or isn't an image.